plugins/secure-mode/src/main/java/ai/djl/serving/plugins/securemode/SecureModeUtils.java

```diff
@@ -55,324 +51,129 @@ public final class SecureModeUtils {
-            Path p = Paths.get(path.trim());
-            if (Files.isDirectory(p)) {
-                if (securityControls.contains(REQUIREMENTS_TXT_CONTROL)) {
-                    scanForDisallowedFile(p, "requirements.txt");
-                }
-                if (securityControls.contains(PICKLE_FILES_CONTROL)) {
-                    scanForPickle(p);
-                }
-                if (securityControls.contains(TRUST_REMOTE_CODE_CONTROL)) {
-                    scanForDisallowedOption(p, "option.trust_remote_code");
-                }
-                if (securityControls.contains(CUSTOM_ENTRYPOINT_CONTROL)) {
-                    scanForDisallowedOption(p, "option.entryPoint");
-                    scanForDisallowedFile(p, "model.py");
-                }
-                if (securityControls.contains(CHAT_TEMPLATE_CONTROL)) {
-                    scanForChatTemplate(p);
-                }
-            } else {
-                throw new IllegalArgumentException("Path " + p + " is not a directory.");
-            }
+        if (securityControls == null) {
+            throw new IllegalConfigurationException(
+                    "Security Controls environment variable is not set.");
         }
-    }
+        logger.info("Secure Mode with the following security controls: {}", securityControls);
 
-    /**
-     * Search for a prohibited file in the directory, and fast-fail if found.
-     *
-     * @param directory the directory to search for the file
-     * @param fileName the name of the file to search for
-     * @throws IOException if there is an error walking the directory
-     * @throws ModelException if a security check fails
-     */
-    private static void scanForDisallowedFile(Path directory, String fileName)
-            throws IOException, ModelException {
-        Path filePath = lookForFile(directory, fileName);
-        if (filePath != null) {
-            throw new ModelException(
-                    "File "
-                            + fileName
-                            + " found at "
-                            + filePath.toString()
-                            + ", but is prohibited in Secure Mode.");
+        String engine = modelInfo.getEngineName();
+        if (!"Python".equals(engine) && !"MPI".equals(engine)) {
+            logger.info("Skip security check for engine: {}", engine);
+            return;
         }
-    }
 
-    /**
-     * Search for any pickle-based file in the directory, and fast-fail if found.
-     *
-     * @param directory the directory to search for pickle-based files
-     * @throws IOException if there is an error walking the directory
-     * @throws ModelException if a security check fails
-     */
-    private static void scanForPickle(Path directory) throws IOException, ModelException {
-        Pattern pattern = Pattern.compile(PICKLE_EXTENSIONS_REGEX);
+        Set<String> controls = new HashSet<>(Arrays.asList(securityControls.split("\\s+,\\s*")));
```

Comment:
I don't understand this regex, doesn't this require a whitespace before the comma? Ideally we just split around commas and then trim whitespace from each member

Reply:
Good catch, fixed.